src/vector.h

```diff
@@ -92,6 +92,9 @@ void git_vector_remove_matching(
 	void *payload);
 
 int git_vector_resize_to(git_vector *v, size_t new_length);
+int git_vector_grow_at(git_vector *v, size_t idx, size_t grow_len);
+int git_vector_shrink_at(git_vector *v, size_t idx, size_t shrink_len);
```

Comment:
Similarly, I think `_grow_at()` is maybe better called `_create_hole()` since we typically use "grow" to mean allocate but not consider it used (but let me know if I'm misremembering)